common/functions/src/scalars/expressions/cast_test.rs

```diff
@@ -47,11 +47,10 @@ fn test_cast_function() -> Result<()> {
 
     for t in tests {
         let rows = t.columns[0].len();
-        let func = t.func;
+        let func = t.func.unwrap();
         if let Err(e) = func.eval(&t.columns, rows) {
             assert_eq!(t.error, e.to_string());
         }
-
         // Display check.
         let expect_display = t.display.to_string();
```

Comment:
need to test the return_type to match the expected data_type

Reply:
And stateless test is needed.

Reply:
All stateless test is passed.